e2e/controls.py

```diff
@@ -2336,3 +2337,134 @@ def expect_n_row(
             n,
             timeout=timeout,
         )
+
+
+### Experimental below
+
+
+class Accordion(
+    # _SetTextM,
+    # _ExpectTextInputValueM,
+    _WidthLocM,
+    _InputWithContainer,
+):
+    # *args: AccordionPanel | TagAttrs,
+    # id: Optional[str] = None,
+    # open: Optional[bool | str | list[str]] = None,
+    # multiple: bool = True,
+    # class_: Optional[str] = None,
+    # width: Optional[CssUnit] = None,
+    # height: Optional[CssUnit] = None,
+    # **kwargs: TagAttrValue,
+    def __init__(self, page: Page, id: str) -> None:
+        super().__init__(
+            page,
+            id=id,
+            loc="> div.accordion-item",
+            loc_container=f"div#{id}.accordion.shiny-bound-input",
+        )
+        self.loc_open = self.loc.locator(
+            # Return self
+            "xpath=.",
+            # Simple approach as position is not needed
+            has=page.locator(
+                "> div.accordion-collapse.show",
+            ),
+        )
+
+    def expect_height(self, value: StyleValue, *, timeout: Timeout = None) -> None:
+        expect_to_have_style(self.loc_container, "height", value, timeout=timeout)
+
+    def expect_width(self, value: StyleValue, *, timeout: Timeout = None) -> None:
+        expect_to_have_style(self.loc_container, "width", value, timeout=timeout)
+
+    def expect_open(
+        self,
+        value: list[PatternOrStr],
+        *,
+        timeout: Timeout = None,
+    ) -> None:
+        playwright_expect(self.loc_open).to_have_count(len(value), timeout=timeout)
+
+    def expect_panels(
+        self,
+        value: list[PatternOrStr],
+        *,
+        timeout: Timeout = None,
+    ) -> None:
+        playwright_expect(self.loc).to_have_count(len(value), timeout=timeout)
+
+    def expect_all_panels_to_have_attribute(
+        self,
+        attribute: str,
+        value: list[PatternOrStr],
+        *,
+        timeout: Timeout = None,
+    ) -> None:
+        # page: Page,
+        # loc_container: Locator,
+        # el_type: str,
+        # arr_name: str,
+        # arr: ListPatternOrStr,
+        # is_checked: bool | MISSING_TYPE = MISSING,
+        # timeout: Timeout = None,
+        # key: str = "value"
+        _MultipleDomItems.expect_locator_values_in_list(
+            page=self.page,
+            loc_container=self.loc_container,
+            el_type="> div.accordion-item",
+            arr_name="value",
+            arr=value,
+            key=attribute,
+            timeout=timeout,
+        )
+
+    def expect_open_panels_to_contain_text(
+        self,
+        value: list[PatternOrStr],
+        *,
+        timeout: Timeout = None,
+    ) -> None:
+        playwright_expect(self.loc_open).to_contain_text(value, timeout=timeout)
```

Comment:
Again, this method feels like it could be covered by `AccordionPanel.expect_body()`.

-------------------------------

What about having a method that returns an `AccordionPanel` as `AccordionPanel`s have a tight association to `Accordion`?

```suggestion
    def accordion_panel(
        self,
        data_value,
    ) -> AccordionPanel:
    	return AccordionPanel(self.page, self.id, data_value)
```